src/ast/codegen_llvm.cpp

```diff
@@ -1923,6 +1959,60 @@ void CodegenLLVM::createFormatStringCall(Call &call, int &id, CallArgs &call_arg
   expr_ = nullptr;
 }
 
+void CodegenLLVM::generateWatchpointSetupProbe(
+    FunctionType *func_type,
+    const std::string &expanded_probe_name,
+    int arg_num,
+    int index)
+{
+  Function *func = Function::Create(func_type,
+                                    Function::ExternalLinkage,
+                                    get_watchpoint_setup_probe_name(
+                                        expanded_probe_name),
+                                    module_.get());
+  func->setSection(
+      get_section_name_for_watchpoint_setup(expanded_probe_name, index));
+  BasicBlock *entry = BasicBlock::Create(module_->getContext(), "entry", func);
+  b_.SetInsertPoint(entry);
+
+  // Send SIGSTOP to curtask
+  b_.CreateSignal(b_.getInt32(SIGSTOP));
```

Comment:
Do we really need to stop the process here? By using this helper this whole change is limited to kernel 5.3. Stopping a process could be risky too.

Reply:
The reason I did the SIGSTOPs is so we can atomically attach the watchpoint when we hit the uprobe. Otherwise anything could happen in between.

SIGSTOP is pretty heavyweight so I wouldn't be against adding a flag to opt out. What do you think that should look like? Maybe an alias like `asyncwatchpoint`? Would rather not add a cmd line flag as it's provider specific.

Reply:
asyncwatchpoint sounds good. If, for some reason, the users wants to send some other signal they can just use signal directly